### PR TITLE
Update HCL & Terraform Lexers

### PR DESCRIPTION
**Pull request by X-Guardian**
This updates the HCL and Terraform Lexers with the following changes:

- Aligns the `id` definition in the HCL Lexer with the Terraform one.
- Moves the `hash` and `array` rules that were added to the Terraform Lexer in #1303 to the HCL Lexer as these changes should apply to the base HCL format.
- Updates the HCL visual sample file with a copy of the Terraform visual sample. This should be highlighted without errors, the same as the Terraform sample, except without the keyword coloring.

### HCL Sample (Before)

Note the red errors

image

### HCL Sample (After)

image

The Terraform sample is identical, before and after these changes.

### CODE REVIEW

**Comment by X-Guardian** [comment]
Hi maintainers, do you need any further detail from me regarding this proposed change? It would be great to get this merged for the next release.

**Comment by tancnle** [comment]
Thank you for your work on this MR, @X-Guardian. I will get to it by the end of the week.

File: lib/rouge/lexers/hcl.rb [hunk 1]
```diff
@@ -123,6 +124,7 @@ def self.builtins
       state :array do
         mixin :comments_and_whitespace
 
+        rule %r/[.,()\\\/*]/, Punctuation
         rule %r/,/, Punctuation
```
Comment (tancnle) [inline lib/rouge/lexers/hcl.rb:128]: I guess the regex on Line 127 already includes `,`. Hence we can drop Line 128. WDYT?

```suggestion
        rule %r/[.,()\\\/*]/, Punctuation
```

Comment (X-Guardian) [reply lib/rouge/lexers/hcl.rb:128]: Yes agreed. I tested without line 128 and the sample hcl renders correctly still.

**Comment by tancnle** [comment]
Thank you for considering the suggestions @X-Guardian. LGTM 🚀

**Comment by tancnle** [comment]
@X-Guardian I will try to cut a new release before the end of this week.

**Comment by X-Guardian** [comment]
That's great. Thanks @tancnle.

**Pull request merged**